src/language/concurrency.md

```diff
@@ -16,90 +19,99 @@ prevpage:
+
+## Event Loop
+
+Dart’s runtime model is based on an event loop. This event loop is responsible
+for executing the code of your program, collecting and processing events,
+running garbage collection processes, and more. As your application runs, all
+events are added to a queue, called the event queue. Events can be anything from
+requests to repaint the UI, to user taps and keystrokes, to i/o from the disk.
+Because your app can’t predict what order these events will happen, it runs this
+event loop that processes events in the order they're queued, one at a time.
 
-## Asynchrony types and syntax
+![A figure showing events being fed, one by one, into the event loop](/assets/img/language/concurrency/event-loop.png)
 
-If you're already familiar with `Future`, `Stream`, and async-await,
-then you can skip ahead to the [isolates section][].
+Implementing an event loop might resemble this:
 
-[isolates section]: #how-isolates-work
+```dart
+while (eventQueue.waitForEvent()) {
+  eventQueue.processNextEvent();
+}
+```
 
+Even though this event loop is synchronous and runs on a single thread, most
+Dart applications need to do more than one thing at a time. For example, a
+client application might need to execute an HTTP request, while also listening
+for a user to tap a button. To handle this, Dart offers many async APIs, like
+Futures, Streams, and async-await. These APIs are built around this event loop.
 
-### Future and Stream types
+For example, consider making a network request:
+```dart
+http.get('https://example.com').then((response) {
+  if (response.statusCode == 200) {
+    print('Success!')'
+  }  
+}
+```
 
-The Dart language and libraries use `Future` and `Stream` objects to
-represent values to be provided in the future.
-For example, a promise to eventually provide an `int` value
-is typed as `Future<int>`.
-A promise to provide a series of `int` values
-has the type `Stream<int>`.
+When this code is processed by the event loop, it will immediately call the
+first clause - `http.get`, and return a `Future`. It will also tell the event
+loop to take note that sometime in the future, a specific event will happen (
+which is in this case an HTTP response), and when that event happens, it should
+execute the callback passed to `then`.
 
-As another example, consider the dart:io methods for reading files.
-The synchronous `File` method [`readAsStringSync()`][]
-reads a file synchronously,
-blocking until the file is either fully read or an error occurs.
-The method then either returns an object of type `String`
-or throws an exception.
-The asynchronous equivalent, [`readAsString()`][],
-immediately returns an object of type `Future<String>`.
-At some point in the future,
-the `Future<String>` completes with either a string value or an error.
+![Figure showing async events being added to an event loop and holding onto 
+a callback to execute later.](/assets/img/language/concurrency/async-event-loop.png)
```

Comment:
The idea that the event loop "holds onto" the callback for later is an illuminating description that you should consider putting in the preceding text.